editor/src/components/canvas/controls/new-canvas-controls.tsx

```diff
@@ -103,13 +107,51 @@ export interface ControlProps {
   maybeClearHighlightsOnHoverEnd: () => void
   transientState: TransientCanvasState
   resolve: ResolveFn
+  propertyTargetOptions: Array<LayoutTargetableProp>
+  propertyTargetSelectedIndex: number
+  setTargetOptionsArray: (newArray: Array<LayoutTargetableProp>) => void
 }
 
 interface NewCanvasControlsProps {
   windowToCanvasPosition: (event: MouseEvent) => CanvasPositions
   cursor: CSSCursor
 }
 
+function useArrayAndIndex(defaultTargets: LayoutTargetableProp[]) {
+  const [targets, setTargets] = React.useState<LayoutTargetableProp[]>(defaultTargets)
+  const [targetIndex, setTargetIndex] = React.useState(0)
+
+  function incrementTargetIndex() {
+    if (targetIndex < targets.length - 1) {
+      setTargetIndex(targetIndex + 1)
+    } else {
+      setTargetIndex(0)
+    }
+  }
+
+  function setTargetsResetIndex(newTargets: LayoutTargetableProp[]) {
+    if (!shallowEqual(targets, newTargets)) {
+      setTargets(newTargets)
+      setTargetIndex(0)
+    }
+  }
```

Comment:
These should be wrapped in `React.useCallback`s

Reply:
Fixed in  5b68dd64a689f218fbe4e203d4db7b5a9433a636.